### ISSUE TITLE
Run in background doesn't work for Flatpak on Phosh

### ISSUE THREAD

**Issue by darkdragon-001** [azymohliad/watchmate#50]
This should be possible to support since [xdg-drsktop-portal 1.16.0](https://www.reddit.com/r/flatpak/comments/zl08c3/xdgdesktopportal_1160_introduces_a_new_background/).

**Comment by azymohliad** [comment]
Hi, thanks for the report. Can you share some more details about your system? Which DE/WM do you use (to understand what's your xdg-desktop-portal backend)?

Also, how exactly does it not work? Does the "Run in background" switch turns back off when you try to turn it on? Or does the switch stays on, but the app then just doesn't keep running in the background when you close the window? Do you have any errors in the output if you run it form the terminal? 

It does perform [the background portal request](https://flatpak.github.io/xdg-desktop-portal/docs/#gdbus-org.freedesktop.portal.Background) when you toggle the switch, so it implements what's needed for a sandboxed app to run in the background. It also works well for me in GNOME 45, so I'm guessing the background request might be handled differently by your xdg-desktop-portal backend.

**Comment by darkdragon-001** [comment]
Thanks for the info. I am running on Pinephone with Mobian using Phosh. The switch turns back off when I try to turn it on. Maybe it's not implemented there?

**Comment by azymohliad** [comment]
Ok, interesting, thanks. I'm not sure. When you have time, could you please try the following?
1. Run it form the terminal, and see if you have "Background portal request failed" error in the output.
```sh
flatpak run io.gitlab.azymohliad.WatchMate
```
2. Manually grant the background permission, check if everything works then:
```sh
flatpak permission-set background background io.gitlab.azymohliad.WatchMate yes
```
3. If no, manually override the background setting, see if it can run in the background after that:
```sh
flatpak enter io.gitlab.azymohliad.WatchMate gsettings --schemadir /app/share/glib-2.0/schemas set io.gitlab.azymohliad.WatchMate run-in-background true
```

**Comment by darkdragon-001** [comment]
> 1. Run it form the terminal, and see if you have "Background portal request failed" error in the output.
> 
> ```shell
> flatpak run io.gitlab.azymohliad.WatchMate
> ```

```
[Error watchmate::ui::settings_page] Background portal request failed: ZBus Error: org.freedesktop.DBus.Error.UnknownMethod: No such interface "org.freedesktop.portal.Background" on object at path /org/freedesktop/portal/desktop
```

> 2. Manually grant the background permission, check if everything works then:
> 
> ```shell
> flatpak permission-set background background io.gitlab.azymohliad.WatchMate yes
> ```

Same problem.

I also checked in Flatseal and the toggle is greyed out there as well.

> 3. If no, manually override the background setting, see if it can run in the background after that:
> 
> ```shell
> flatpak enter io.gitlab.azymohliad.WatchMate gsettings --schemadir /app/share/glib-2.0/schemas set io.gitlab.azymohliad.WatchMate run-in-background true
> ```

I could run this while the app was running, but when I closed the window, the process stopped. Don't know how the run in background should work or what Phosh is doing when one swipes a window up to close.

Thanks for your help anyways. Probably I just have to wait for Phosh to implement the necessary functionality. Feel free to close if you don't have any further ideas.

**Comment by azymohliad** [comment]
Ok, so the portal isn't implemented on Phosh (yet?). Still, I think Watchmate should actually allow to toggle the setting if the portal isn't implemented, because the permission might not be needed then. And I just realized, the 3rd step I suggested doesn't actually toggle it properly at runtime. Could you please try if [this](https://mega.nz/file/hUdhHIgI#kSf_6rW8tB215hCDQcy1rRxttGyfZBiDg_UUYqR8qbU) build helps? If yes, I'll release v0.5.1 with the fix soon, probably tomorrow

**Comment by azymohliad** [comment]
Ouch, sorry, i'm dumb. I just built it for x86, completely forgot we're talking about the PinePhone. I'll cross-compile it on Flathub CI later today

**Comment by azymohliad** [comment]
Didn't manage it yesterday, but [here's](https://github.com/flathub/io.gitlab.azymohliad.WatchMate/pull/10#issuecomment-1802129800) the Flathub CI build that enables the background mode setting even if background portal is not implemented:
```
flatpak install --user https://dl.flathub.org/build-repo/62591/io.gitlab.azymohliad.WatchMate.flatpakref
```

If you could try it whenever you have time, I'd be grateful!

**Comment by darkdragon-001** [comment]
Seems to work (I can activate the setting, when I close the window the process continues and when I run it again then the window disappears and the second process finishes immediately as it only triggers the window on the first process)! Thanks a lot!

**Comment by azymohliad** [comment]
Nice, thank you for your help! I'll release it later today

**Comment by azymohliad** [comment]
I released v0.5.1, it should be on Flathub in a few hours
